# Deep occurrence-support XGBoost screen

**Decision:** retain occurrence-free archived deep XGBoost. Category support weakens both the component and its fixed archive substitution.

In [1]:
import json
from pathlib import Path
import pandas as pd

STAGE_DIR = Path.cwd().resolve().parent if Path.cwd().name == 'notebooks' else Path.cwd().resolve()
OUTPUT_DIR = STAGE_DIR.parent / '.runtime' / 'deep-occurrence-xgboost-screen'
result = json.loads((OUTPUT_DIR / 'result.json').read_text(encoding='utf-8'))
summary = pd.read_csv(OUTPUT_DIR / 'candidate-summary.csv', index_col=0)
diversity = pd.read_csv(OUTPUT_DIR / 'diversity.csv', index_col=[0, 1])
result

{'occurrence_fields': 29,
 'minimum_support': 5,
 'occurrence_deep_accuracy': 0.813047138047138,
 'archive_accuracy': 0.8178661616161615,
 'archive_change': -6.313131313140374e-05,
 'passes_gate': False,
 'local_test_opened': False,
 'competition_predictions_generated': False}

## Course-aligned lifecycle

1. **Define the goal and scope** — test whether occurrence support complements top-common identities.
2. **Gather the data** — reuse supplied predictors and frozen development rows.
3. **Explore the data** — cross two useful archive signals without opening a grid.
4. **Clean and preprocess the data** — fit support inside each fold and map counts below five to the rare marker.
5. **Select and engineer features** — add support for all 29 categorical fields.
6. **Define the machine-learning task** — retain three-class accuracy.
7. **Partition the data** — reuse the five frozen stratified folds.
8. **Select and train candidate methods** — fit one predeclared archived depth-17 XGBoost seed.
9. **Evaluate and interpret the results** — compare the component and fixed archive substitution.
10. **Deploy and iterate** — stop before local or competition use because accuracy weakens.

In [2]:
display(summary.style.format({
    'mean_accuracy': '{:.4%}',
    'incumbent_change': '{:+.4%}',
    'worst_fold_change': '{:+.4%}',
    'repair_recall_change': '{:+.4%}',
}))
diversity.loc[[
    ('XGBoost archived depth 17 [top-50 deferred identities; seed 20260822]',
     'XGBoost archived depth 17 [top-50 identities plus occurrence support; seed 20260822]')
]]

,mean_accuracy,incumbent_change,fold_wins_vs_incumbent,worst_fold_change,repair_recall_change,passes_gate
candidate,,,,,,
existing_deep_archive_substitution,81.8771%,+0.0842%,3,-0.0210%,+0.6945%,False
existing_archive_synthesis,81.7929%,+0.0000%,0,+0.0000%,+0.0000%,False
occurrence_support_archive_substitution,81.7866%,-0.0063%,3,-0.1578%,+0.8391%,False
archived_deep_xgboost,81.4710%,-0.3220%,0,-0.5892%,+0.5207%,False
occurrence_support_deep_xgboost,81.3047%,-0.4882%,0,-0.6629%,+1.5626%,False


,,disagreement,left_only_correct,right_only_correct,both_wrong
left_model,right_model,,,,
XGBoost archived depth 17 [top-50 deferred identities; seed 20260822],XGBoost archived depth 17 [top-50 identities plus occurrence support; seed 20260822],0.043245,0.020476,0.018813,0.166477


## Conclusion

The occurrence-support component falls from **81.4710%** to **81.3047%**. Its archive substitution reaches **81.7866%**, 0.0063 percentage points below the archive incumbent. The branch is closed without opening the local test or generating competition predictions.